modules/packages/LinearAlgebra.chpl

```diff
@@ -221,6 +222,260 @@ class LinearAlgebraError : Error {
+      isExactdComputed[1] = true;
+    }
+    return this.Exactd4;
+  }
+
+  proc comp_Exactd6(){
+    if !isExactdComputed[2] then {
+      this.Exactd6 = norm(comp_A6(), normType.norm1)**(1/6.0);
+      isExactdComputed[2] = true;
+    }
+    return this.Exactd6;
+  }
+
+  proc comp_Exactd8(){
+    if !isExactdComputed[3] then {
+      this.Exactd8 = norm(comp_A8(), normType.norm1)**(1/8.0);
+      isExactdComputed[3] = true;
+    }
+    return this.Exactd8;
+  }
+
+  proc comp_Exactd10(){
+    if !isExactdComputed[4] then {
+      this.Exactd10 = norm(comp_A10(), normType.norm1)**(1/10.0);
+      isExactdComputed[4] = true;
+    }
+    return this.Exactd10;
+  }
+
+  proc comp_Loosed4(){
+    if this.use_exact_onenorm then{
+      return comp_Exactd4();
+    }
+    else {
+      if !isApproxdComputed[1] then {
+        //TODO: Need to write a function to compute the
+        // estimated norm (change the below function)
+        // Right now, just returns the exact norm
+        this.Approxd4 = norm(comp_A4(), normType.norm1)**(1/4.0);
+        isApproxdComputed[1] = true;
+      }
+      return this.Approxd4;
+    }
+  }
+
+  proc comp_Loosed6(){
+    if this.use_exact_onenorm then{
+      return comp_Exactd6();
+    }
+    else {
+      if !isApproxdComputed[2] then {
+        //TODO: Need to write a function to compute the
+        // estimated norm (change the below function)
+        // Right now, just returns the exact norm
+        this.Approxd6 = norm(comp_A6(), normType.norm1)**(1/6.0);
+        isApproxdComputed[2] = true;
+      }
+      return this.Approxd6;
+    }
+  }
+
+  proc comp_Loosed8(){
+    if this.use_exact_onenorm then{
+      return comp_Exactd8();
+    }
+    else {
+      if !isApproxdComputed[3] then {
+        //TODO: Need to write a function to compute the
+        // estimated norm (change the below function)
+        // Right now, just returns the exact norm
+        this.Approxd8 = norm(comp_A8(), normType.norm1)**(1/8.0);
+        isApproxdComputed[3] = true;
+      }
+      return this.Approxd8;
+    }
+  }
+
+  proc comp_Loosed10(){
+    if this.use_exact_onenorm then{
+      return comp_Exactd10();
+    }
+    else {
+      if !isApproxdComputed[4] then {
+        //TODO: Need to write a function to compute the
+        // estimated norm (change the below function)
+        // Right now, just returns the exact norm
+        this.Approxd10 = norm(comp_A10(), normType.norm1)**(1/10.0);
+        isApproxdComputed[4] = true;
+      }
+      return this.Approxd10;
+    }
+  }
+
+  proc pade3(){
+    var b = [120.0, 60.0, 12.0, 1.0];
+    var U = matrix_prod(this.A, reshape(b[3]*comp_A2() + b[1]*this.ident, this.A.indices));
+    var V = reshape(b[2]*comp_A2() + b[0]*this.ident, this.A.indices);
+    return (U,V);
+  }
+
+  proc pade5(){
+    var b = [30240.0, 15120.0, 3360.0, 420.0, 30.0, 1.0];
+    var U = matrix_prod(this.A, reshape(b[5]*this.A4 + b[3]*this.A2 + b[1]*this.ident, this.A.indices));
```

Comment:
The code here and below doesn't use `comp_A4`, but `this.A4`. I think you need `comp_A4` and friends below.

However, a different suggestion here -- it's a little hard to figure out whether this ends up creating a temporary array. So I might suggest that the `comp_A?` functions not actually return anything, but cache the results.

Reply:
Yes, I'm not sure why have I used this.A4 and not comp_Ax method.

Yeah, so the intent of comp_Ax methods is "to only compute Ax if necessary else don't compute it". Had we statically defined something like 
```
A4 = dot(A2, A2)
A6 = dot(A4, A2)
.
.
```
It would compute these matrices even if not needed. On the contrary, do you think it's overkill?